app/models/manageiq/providers/hawkular/middleware_manager.rb

```diff
@@ -203,7 +203,7 @@ def self.update_alert_profile(*args)
         :new_assignments_ids => process_new_assignments_ids(alert_profile_arg[:new_assignments])
       }
```

Comment:
What about?

```ruby
miq_alert_profile = {
  :id                  => alert_profile_arg[:profile_id],
  :old_alerts_ids      => alert_profile_arg[:old_alerts],
  :new_alerts_ids      => alert_profile_arg[:new_alerts],
  :old_assignments_ids => process_old_assignments_ids(alert_profile_arg[:old_assignments]),
  :new_assignments => { 
    :ids           => process_new_assignments_ids(alert_profile_arg[:new_assignments]),
    :resource_type => alert_profile_arg[:new_assignments]["assign_to"]
  }
}
```
Otherwise, if the `resource_type` is common between old assignments and new assignments, it could be moved to the hash root:

```ruby
miq_alert_profile = {
  :id                  => alert_profile_arg[:profile_id],
  :old_alerts_ids      => alert_profile_arg[:old_alerts],
  :new_alerts_ids      => alert_profile_arg[:new_alerts],
  :resource_type       => alert_profile_arg[:new_assignments]["assign_to"],
  :old_assignments_ids => process_old_assignments_ids(alert_profile_arg[:old_assignments]),
  :new_assignments_ids => process_new_assignments_ids(alert_profile_arg[:new_assignments])
}
```

Adding the resource_type into the hash, you keep using the idea of wrap all the data into one param (kinda DTO) and, as well, you keep leaving one basic param which is pretty determing: the operation.

Just an idea.

Reply:
@xeviknal I like this, and I'd go even further, why not have it as a value object, like this: https://github.com/ManageIQ/manageiq-providers-hawkular/blob/master/app/models/manageiq/providers/hawkular/inventory/server_operations.rb#L35

You'd pass alert_profile_arg as an argument and have all those keys as methods.

Reply:
Great. This object could handle all the logics related to the operation.
What I am not sure is if it is now the moment to do the refactor or made it aside.

Reply:
Imho it is best to do this on this PR, to mantain the complexity scale. It should be done in one commit that we can revert if necessary, but I think doing it now is better than doing it later.

Reply:
Changed